common/validations.rb

```diff
@@ -143,6 +201,172 @@ def self.check_date(hash)
     errors
   end
 
+  def self.check_structured_date_label(hash)
+    errors = []
+
+    if hash["structured_date_range"] && hash["structured_date_single"]
+      errors << ["structured_date_single", "cannot specify both a single and ranged date"]
+    end
+
+    if !hash["structured_date_range"] && !hash["structured_date_single"]
+      errors << ["structured_date_single", "must specifiy either a single or ranged date"]
+    end
+
+    if hash["structured_date_range"] && hash["date_type_enum"] == "single"
+      errors << ["structured_date_range", "Must specify single date for date type of single"]
+    end
+
+    if hash["structured_date_single"] && hash["date_type_enum"] == "range"
+      errors << ["structured_date_range", "Must specify range date for date type of range"]
+    end
+
+    return errors
+  end
+
+  def self.check_structured_date_single(hash)
+    errors = []
+
+    if hash["date_role_enum"].nil?
+      errors << ["date_role_enum", "is required"]
+    end
+
+    has_expr_date = !hash["date_expression"].nil? && 
+                    !hash["date_expression"].empty?
+
+    has_std_date = !hash["date_standardized"].nil? 
+
+    errors << ["date_standardized", "or date expression is required"] unless has_expr_date || has_std_date
+
+    if has_std_date
+      matches_y          = (hash["date_standardized"] =~ /^[\d]{1}$/) == 0
```

Comment:
@avatar382 I wonder if this shouldn't be a util method since it is used more than once?